src/box/raft.c

```diff
@@ -952,12 +952,19 @@ raft_cfg_is_candidate(bool is_candidate)
 			 * until the new state is fully persisted.
 			 */
 		}
-	} else if (raft.state != RAFT_STATE_FOLLOWER) {
-		if (raft.state == RAFT_STATE_LEADER)
-			raft.leader = 0;
-		raft.state = RAFT_STATE_FOLLOWER;
-		/* State is visible and changed - broadcast. */
-		raft_schedule_broadcast();
+	} else {
+		if (raft.state != RAFT_STATE_LEADER) {
+			/* Do not wait for anything while being a voter. */
+			assert(ev_is_active(&raft.timer));
```

Comment:
This may crash, if there is an active WAL-write task.  I would just stop the timer unconditionally (if it is implemented as nop in case it is already stopped).

Reply:
Agree, fixed.
It is implemented this way, indeed.

https://github.com/tarantool/tarantool/blob/sp/raft-crash-on-voter/third_party/libev/ev.c#L3991-L3997